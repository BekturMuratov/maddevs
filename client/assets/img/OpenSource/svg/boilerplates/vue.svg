<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>