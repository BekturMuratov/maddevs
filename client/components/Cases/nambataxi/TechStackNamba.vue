<template>
  <section class="container_regular">
    <h2 class="case_title_h2 m-24_bottom media-m-12_bottom case_technology-title">
      Technologies we used
    </h2>
    <ListTechnologies class="m-96_bottom media-m-48_bottom">
      <ListTechnologiesItem
        v-for="technology in techStackNambaTaxi"
        :key="technology.name"
        class="case_tech-list-item"
        v-bind="technology"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import ListTechnologies from '@/components/Cases/shared/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { techStackNambaTaxi } from '@/data/caseNambaTaxi'

export default {
  name: 'TechStackNamba',
  components: {
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      techStackNambaTaxi,
    }
  },
}
</script>

<style lang="scss" scoped>
.case {
  &_technology-title{
    text-align: center;
  }
  &_tech-list-item::v-deep {
    .django {
      width: 57px;
    }
    .osrm, .django-rest{
      width: 63px;
    }

    @media screen and (max-width: 768px) {
      p {
        font-size: 8px;
      }
      .django, .osrm {
        width: 30px;
      }
    }
  }
}
</style>
