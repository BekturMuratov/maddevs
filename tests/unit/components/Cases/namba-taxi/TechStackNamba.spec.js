import { render, screen } from '@testing-library/vue'
import TechStackNamba from '@/components/Cases/nambataxi/TechStackNamba'

const stubs = ['ListTechnologiesItem']

describe('TechStackNamba component', () => {
  it('should render correctly', () => {
    const { container } = render(TechStackNamba, {
      stubs,
    })
    expect(container).toMatchSnapshot()
  })

  it('title should include classname', () => {
    render(TechStackNamba, {
      stubs,
    })
    expect(screen.getByText(/Technologies we used/i).className).toContain('case_title_h2')
  })
})
