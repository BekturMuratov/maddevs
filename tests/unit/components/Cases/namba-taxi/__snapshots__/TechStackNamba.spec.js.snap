// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechStackNamba component should render correctly 1`] = `
<div>
  <section
    class="container_regular"
  >
    <h2
      class="case_title_h2 m-24_bottom media-m-12_bottom case_technology-title"
    >
      
    Technologies we used
  
    </h2>
     
    <ul
      class="case_list-technologies m-96_bottom media-m-48_bottom"
    >
      <listtechnologiesitem-stub
        alt="Python"
        class="case_tech-list-item"
        imgalt="Python"
        name="python"
      />
      <listtechnologiesitem-stub
        alt="Django"
        class="case_tech-list-item"
        imgalt="Django"
        name="django"
      />
      <listtechnologiesitem-stub
        alt="Django REST"
        class="case_tech-list-item"
        imgalt="Django REST"
        name="django-rest"
      />
      <listtechnologiesitem-stub
        alt="Celery"
        class="case_tech-list-item"
        imgalt="Celery"
        name="celery"
      />
      <listtechnologiesitem-stub
        alt="Go"
        class="case_tech-list-item"
        imgalt="Go"
        name="go"
      />
      <listtechnologiesitem-stub
        alt="Jinja"
        class="case_tech-list-item"
        imgalt="Jinja"
        name="jinja"
      />
      <listtechnologiesitem-stub
        alt="uWSGI"
        class="case_tech-list-item"
        imgalt="uWSGI"
        name="uwsgi"
      />
      <listtechnologiesitem-stub
        alt="Nginx"
        class="case_tech-list-item"
        imgalt="Nginx"
        name="nginx"
      />
      <listtechnologiesitem-stub
        alt="HAProxy"
        class="case_tech-list-item"
        imgalt="HAProxy"
        name="haproxy"
      />
      <listtechnologiesitem-stub
        alt="Docker"
        class="case_tech-list-item"
        imgalt="Docker"
        name="docker"
      />
      <listtechnologiesitem-stub
        alt="Docker Compose"
        class="case_tech-list-item"
        imgalt="Docker Compose"
        name="docker-compose"
      />
      <listtechnologiesitem-stub
        alt="GitLab CI/CD"
        class="case_tech-list-item"
        imgalt="GitLab CI/CD"
        name="gitlab"
      />
      <listtechnologiesitem-stub
        alt="Jenkins"
        class="case_tech-list-item"
        imgalt="Jenkins"
        name="jenkins"
      />
      <listtechnologiesitem-stub
        alt="Sentry"
        class="case_tech-list-item"
        imgalt="Sentry"
        name="sentry"
      />
      <listtechnologiesitem-stub
        alt="Grafana"
        class="case_tech-list-item"
        imgalt="Grafana"
        name="grafana"
      />
      <listtechnologiesitem-stub
        alt="Graylog"
        class="case_tech-list-item"
        imgalt="Graylog"
        name="graylog"
      />
      <listtechnologiesitem-stub
        alt="Kibana"
        class="case_tech-list-item"
        imgalt="Kibana"
        name="kibana"
      />
      <listtechnologiesitem-stub
        alt="MySQL"
        class="case_tech-list-item"
        imgalt="MySQL"
        name="mysql"
      />
      <listtechnologiesitem-stub
        alt="Redis"
        class="case_tech-list-item"
        imgalt="Redis"
        name="redis"
      />
      <listtechnologiesitem-stub
        alt="Elasticsearch"
        class="case_tech-list-item"
        imgalt="Elasticsearch"
        name="elastic"
      />
      <listtechnologiesitem-stub
        alt="MinIO"
        class="case_tech-list-item"
        imgalt="MinIO"
        name="minio"
      />
      <listtechnologiesitem-stub
        alt="OpenStreetMap"
        class="case_tech-list-item"
        imgalt="OpenStreetMap"
        name="open-street-map"
      />
      <listtechnologiesitem-stub
        alt="OSRM"
        class="case_tech-list-item"
        imgalt="OSRM"
        name="osrm"
      />
      <listtechnologiesitem-stub
        alt="gRPC"
        class="case_tech-list-item"
        imgalt="gRPC"
        name="grpc"
      />
      <listtechnologiesitem-stub
        alt="Node.js"
        class="case_tech-list-item"
        imgalt="Node.js"
        name="node"
      />
      <listtechnologiesitem-stub
        alt="Twisted"
        class="case_tech-list-item"
        imgalt="Twisted"
        name="twisted"
      />
      <listtechnologiesitem-stub
        alt="Ruby"
        class="case_tech-list-item"
        imgalt="Ruby"
        name="ruby"
      />
      <listtechnologiesitem-stub
        alt="WebRTC"
        class="case_tech-list-item"
        imgalt="WebRTC"
        name="webrtc"
      />
      <listtechnologiesitem-stub
        alt="HTML"
        class="case_tech-list-item"
        imgalt="HTML"
        name="html"
      />
      <listtechnologiesitem-stub
        alt="CSS"
        class="case_tech-list-item"
        imgalt="CSS"
        name="css"
      />
      <listtechnologiesitem-stub
        alt="jQuery"
        class="case_tech-list-item"
        imgalt="jQuery"
        name="jquery"
      />
      <listtechnologiesitem-stub
        alt="Bootstrap"
        class="case_tech-list-item"
        imgalt="Bootstrap"
        name="bootstrap"
      />
      <listtechnologiesitem-stub
        alt="JavaScript"
        class="case_tech-list-item"
        imgalt="JavaScript"
        name="js"
      />
      <listtechnologiesitem-stub
        alt="React"
        class="case_tech-list-item"
        imgalt="React"
        name="react"
      />
    </ul>
  </section>
</div>
`;
